File: works/tmp/1138.py
from itertools import permutations

N = int(input())
order = map(int, input().split())
order = list(order)

line = []
for i in range(1, N+1): line.append(i)

# line = list(permutations(line, len(line)))
line = permutations(line, len(line))

for l in line:
    isAnswer = True

    for i in range(0, N):
        cur = l[i]
        taller = order[cur-1]
        cnt = 0

        for j in range(0, i):
            if l[j] > l[i]: cnt+=1

        if cnt != taller:
            isAnswer = False
            break

    if isAnswer:
        for i in range(0, N): print(l[i], end=' ')
        break

File: works/tmp/1138.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;
using static System.Console;
using System.Text;

namespace ConsoleApp
{
    class Program
    {
        static int N;
        static bool fin = false;

        static void Permutation(int[] line, bool[] mark, List<int> l, int depth, int[] order)
        {
            if (depth == N)
            {
                bool isAnswer = true;
                for (int i = 1; i <= N; i++)
                {
                    int cur = l[i];
                    int taller = order[cur];
                    int cnt = 0;
                    for (int j = 1; j < i; j++)
                    {
                        if (l[j] > l[i]) cnt++;
                    }
                    if (cnt != taller) { isAnswer = false; break; }
                }
                if (isAnswer)
                {
                    for (int i = 1; i <= N; i++) Write(l[i] + " ");
                    fin = true;
                }
                return;
            }

            for (int i = 1; i <= N; i++)
            {
                if (mark[i]) continue;
                mark[i] = true;
                l.Add(line[i]);
                Permutation(line, mark, l, depth + 1, order);
                mark[i] = false;
                l.RemoveAt(l.Count - 1);
            }
        }

        static void Main(string[] args)
        {
            StreamReader sr = new StreamReader(Console.OpenStandardInput());
            N = int.Parse(sr.ReadLine());

            int[] line = new int[N + 1];
            for (int i = 1; i <= N; i++) line[i] = i;

            string[] _s = sr.ReadLine().Split(' ');
            int[] order = new int[N + 1];
            for (int i = 1; i <= N; i++) order[i] = int.Parse(_s[i - 1]);

            bool[] mark = new bool[N + 1];

            List<int> l = new List<int>();
            l.Add(0);
            Permutation(line, mark, l, 0, order);
        }

    }
}
